tests/testthat/test_ggetbout.R

```diff
@@ -60,32 +60,87 @@ test_that("g.getbout produces expected output", {
-  bm6Ex = c(rep(0,27), rep(1,10), rep(0,20))
-  bm6Eboutcount = c(rep(0,27), rep(1,10), rep(0,20))
-  expect_that(sum(bm6$x),equals(10))
-  # Metric 6 - ability to detect even length bout
-  set.seed(300)
-  xtest = x = c(rep(1, 4), rep(0, 3), rep(1, 6),rep(0, 3), rep(1, 10), 0 , rep(1, 9))
-  bm6b = g.getbout(x=xtest,boutduration = 10, boutcriter=1,
-                  closedbout=FALSE,bout.metric=6,ws3=60)
-  expect_that(sum(bm6b$x),equals(10))
-  expect_that(sum(bm6b$boutcount),equals(10))
-  
-  # Metric 6 - ability to detect odd length bouts
-  set.seed(300)
-  xtest = x = c(rep(0, 20), rep(1, 9), rep(0, 20))
-  bm6c = g.getbout(x=xtest,boutduration =9, boutcriter=1,
-                   closedbout=FALSE,bout.metric=6,ws3=60)
-  expect_that(sum(bm6c$x),equals(9))
-  # Metric 6 - ability to detect bouts with grace periods
-  xtest = c(rep(0,15), rep(1,6), 0, rep(1,10),rep(0,20))
-  bm6 = g.getbout(x=xtest,boutduration = 10, boutcriter=0.9,
-                  closedbout=FALSE,bout.metric=6,ws3=60)
-  expect_that(sum(bm6$x),equals(17))
+  # -------------
+  # Metric 6 - function to create fake bouts of a given duration and boutcrit
+  # in this way, we can predict the time in bouts that should be detected
+  # -------------
+  dummybout = function(totaldur_min = 60,
+                       ws3 = 5,
+                       boutdur = 10,
+                       boutdur_detect = 10,
+                       boutcriter = 0.8,
+                       nBouts = 1) {
+    x = rep(0, totaldur_min*60/ws3) # initialize vector fo 1 hour
+    boutdur = boutdur * 60/ws3
+    boutdur_detect = boutdur_detect * 60/ws3
+    gaps = boutdur_detect * (1-boutcriter)
+    bout = rep(1, boutdur)
+    # random position for gaps without gaps > 1min
+    gap1min = 60/ws3 + 1
+    longest_gap = gap1min + 1
+    while (longest_gap >= gap1min) {
+      zeroes = sort(sample(x = 2:(length(bout) - 1), size = gaps))
+      temp <- cumsum(c(1, diff(zeroes) - 1))
+      temp2 <- rle(temp)
+      longest_gap_tmp = zeroes[which(temp == with(temp2, values[which.max(lengths)]))]
+      longest_gap = length(longest_gap_tmp)
+      # if longest_gap <= 1min, then the random zeroes generated are fine and this breaks the loop
+    }
+    bout[zeroes] = 0
+    # random position in x to place the bouts
+    sep_between_bouts = boutdur + gap1min #bouts need to be separated at least for more than 1 min
+    sep = sep_between_bouts - 1
+    while (sep < sep_between_bouts) {
+      where = sort(sample(x = 1:(length(x) - boutdur), size = nBouts))
+      if (length(where) < 2) sep = sep_between_bouts + 1 # if only one bout, any position is good
+      if (length(where) >= 2) {
+        sep_bouts = diff(where)
+        sep = min(sep_bouts)
+      }
+    }
+    # impute bouts in x
+    for (start_i in where) x[start_i:(start_i + boutdur - 1)] = bout
+    
+    return(x)
+  }
   
+  # -------------
+  # Metric 6 - random tests with variable configurations
+  # -------------
+  for (i in 1:500) {
```

Comment:
Random unit tests without setting a seed is not practical. If a unit test fails we will not know what caused it to fail, because each run will be different. Instead, use set.seed to make the random process deterministic, or even better test a range of specific scenario such that we always know for sure that those are tested.